quantile/cm/heap_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMinHeapPushInDecreasingOrder(t *testing.T) {
+	h := &minHeap{}
+	iter := 10
+	for i := iter - 1; i >= 0; i-- {
+		h.Push(float64(i))
+		require.Equal(t, iter-i, h.Len())
+	}
+	for i := 0; i < iter; i++ {
+		require.Equal(t, float64(i), h.Min())
+		require.Equal(t, float64(i), h.Pop())
+	}
+}
```

Comment:
Might be worth adding a few more min heap tests?  We could potentially take the cases already written for the golang stdlib heap and just simply write up equivalent cases.  It should be pretty easy to translate them I should think?  Don't need all of them but having good coverage would be ideal.